docs/server/source/dev-and-test/index.rst

```diff
@@ -5,4 +5,4 @@ Develop & Test BigchainDB Server
    :maxdepth: 1
 
    setup-run-node
-   running-unit-tests
+   running-all-tests
```

Comment:
Hmm, I think this section isn't so important (especially now that we're changing its link), but let's keep it for now.

I'm of the opinion that we don't need to add too much dev documentation in the readthedocs, since that's what our repo and Github is for.